File: python/7 kyu/33.py
# https://www.codewars.com/kata/5eb34624fec7d10016de426e/train/python

# Passed

def martingale(bank, outcomes):
    stake = 100
    for outcome in outcomes:
        win = outcome
        if win:
            bank += stake
            stake = 100
        else:
            bank -= stake
            stake *= 2 
    
    return bank

output = martingale(1000, [1, 1, 0, 0, 1])
print(output)

File: nodejs/7 kyu/33.js
// https://www.codewars.com/kata/558f9f51e85b46e9fa000025/train/javascript

// Passed

function differenceOfSquares(n) {
  const numbersTillN = [];
  for (let i = 0; i < n; i++) {
    numbersTillN.push(i + 1);
  }

  return (
    Math.pow(
      numbersTillN.reduce((prev, curr) => prev + curr, 0),
      2
    ) -
    numbersTillN
      .map((num) => Math.pow(num, 2))
      .reduce((prev, curr) => prev + curr, 0)
  );
}

const output = differenceOfSquares(10);
console.log(output);
